packages/gatsby-transformer-code-examples/gatsby-node.js

```diff
@@ -0,0 +1,74 @@
+const createSchemaCustomization = ({ actions, schema }) => {
+  const { createTypes } = actions;
+
+  const typeDefs = [
+    schema.buildObjectType({
+      name: 'CodeExample',
+      fields: {
+        name: 'String!',
+        extension: 'String!',
+        absolutePath: 'String!',
+        content: 'String!',
+      },
+      interfaces: ['Node'],
+      extensions: {
+        infer: false,
+      },
+    }),
+  ];
+
+  createTypes(typeDefs);
+};
+
+async function onCreateNode({
+  node,
+  actions,
+  loadNodeContent,
+  createNodeId,
+  createContentDigest,
+}) {
+  const validExtensions = [
+    'application/javascript',
+    'text/x-java-source',
+    'application/octet-stream',
+    'application/json',
+    'application/x-httpd-php',
+    'application/x-sh',
+    'video/mp2t',
+    'text/yaml',
+  ];
+  if (!validExtensions.includes(node.internal.mediaType)) return;
+
+  const { createNode, createParentChildLink } = actions;
+
+  function transformObject(obj, id, type) {
+    const codeSampleNode = {
+      ...obj,
+      id,
+      children: [],
+      parent: node.id,
+      internal: {
+        contentDigest: createContentDigest(obj),
+        type,
+      },
+    };
+    createNode(codeSampleNode);
+    createParentChildLink({ parent: node, child: codeSampleNode });
+  }
+
+  const content = await loadNodeContent(node);
+
+  transformObject(
+    {
+      name: node.name,
+      extension: node.extension,
+      absolutePath: node.absolutePath,
```

Comment:
If you choose to allow a prefix, then the path should rather be the relative path.  I would prefer to not work with absolute paths at all because they can be missing or totally different strings in source plugins that are not filesystem (but still provide the content, e.g. the git plugin.